File: py/2782.py
n = int(input())
sequence = [int(i) for i in input().split()]

stepladders = 1
for i in range(1, n - 1):
    if sequence[i] - sequence[i - 1] != sequence[i + 1] - sequence[i]:
        stepladders += 1
        
print(stepladders)

File: cpp/2782.cpp
#include <iostream>

using namespace std;

int main()
{
    int n, i, stepladders = 1;
    cin >> n;
    int sequence[n];

    for (i = 0; i < n; i++) {
        cin >> sequence[i];
    }

    for (i = 1; i < n - 1; i++) {
        if (sequence[i] - sequence[i - 1] != sequence[i + 1] - sequence[i]) {
            stepladders++;
        }
    }

    cout << stepladders << '\n';
    return 0;
}